### ISSUE TITLE
iCE40-HX8K Breakout Board (iCE40HX-8k CT256 device)

### ISSUE THREAD

**Issue by ozzyrob** [multigcs/LinuxCNC-RIO#3]
Not an exactly an issue just reaching out.
Saw your thread on the Linuxcnc forum (can't use the forum but that is another story), anyways I thought I had a tiny BX board but alas not the case, but I do have one of these, [iCE40-HX8K Breakout Board ](https://www.latticesemi.com/Products/DevelopmentBoardsAndKits/iCE40HX8KBreakoutBoard.aspx) . Plus a couple of 3.3 to 5v breakout boards I used with a beagle bone black.
As this particular board has a 12MHz clock and a USB to SPI programmer, could I use the config TurBos uses for the Icebreaker as a basis ?
Sorry if this is the wrong channel to reach out. I hope it was ok to upload the info for this board.
[FPGA-EB-02031-1-2-iCE40HX-8K-Breakout-Board-User-Guide.pdf](https://github.com/multigcs/LinuxCNC-RIO/files/11623247/FPGA-EB-02031-1-2-iCE40HX-8K-Breakout-Board-User-Guide.pdf)

**Comment by multigcs** [comment]
Hi, you can use:
configs/Olimex-ICE40HX8K-EVB_BOB/config.json
as basis add adding the osc parameter:
    "clock": {
        "osc": "12000000",
        "speed": "100000000",
        "pin": "J3"
    },

**Comment by ozzyrob** [comment]
That's fantastic.
Just working on a spreadsheet to get my head around FPGA pin mapping to header pins, once that's sorted I'll setup the RPi 400 and start messing around.

**Comment by multigcs** [comment]
# While I'm asking...... will there be an option for an index pulse on an encoder, say for threading for a lathe ?
not supported yet, because i have no CNC-Lathe and i don't know what i should do with the index-pin .

i will check this...

**Comment by ozzyrob** [comment]
Sorry for so long to get back. Spent ages setting up OS for getting going, only to find out that Debian Bookworm doesn't want to play nice with SPI. Found a user on the Linuxcnc forum that had a prebuilt image based on Bullseye with Linuxcnc installed.
Everything set and the Lattice Eval board works. Had an issue with SPI when using Dupoint jumpers, solved that with a little hacked together PCB. Getting problem free SPI communication, with the divisor at 128.
Really enjoying this and appreciating your work, thank you sir.
Next is to hook up some 3.3v to 5v breakout boards I used with my Beaglebone Black & get start testing the Z axis on the ML7/Super 7 Fankenstein thing in the garage.
Cheers Rob
